reconcile/test/test_utils_github_api.py

```diff
@@ -0,0 +1,18 @@
+from unittest import TestCase
+from unittest.mock import patch
+from requests.exceptions import ConnectTimeout
+from reconcile.utils.github_api import GithubApi
+
+
+class TestGithubApi(TestCase):
+
+    @patch("reconcile.utils.github_api.GH_BASE_URL", "http://198.18.0.1")
+    def test_github_client_timeout(self):
+        with patch('reconcile.utils.github_api.SecretReader', autospec=True) \
```

Comment:
I tend to dislike using this `with` format, any changes to indentation will make your tests use the real class and it'll be very confusing. Why don't you decorate this method with the new base URL and the `SecretReader`?

Reply:
I think this concern applies in a more general python level, or any other languages that manage the code structures with indents (like nested loops, etc). 
Changed to a decorator flavour.